lib/livebook_web/templates/error/401_secret.html.eex

```diff
@@ -17,12 +17,18 @@
           Authentication required
         </div>
         <div class="max-w-2xl text-center text-gray-300">
-          Please check out the console for authentication URL
-          or type the token directly here.
+          <%= if @type == :token do %>
+            Please check out the console for authentication URL
+            or type the token directly here.
+          <% end %>
+          <%= if @type == :password do %>
+            Type password to access the live book.
+          <% end %>
         </div>
         <div class="text-2xl text-gray-50 w-full pt-2">
-          <form method="get" class="flex flex-col space-y-4 items-center">
-            <input type="text" name="token" class="input" placeholder="Token" />
+          <form method="<%= if @type == :token, do: "get", else: "post" %>" class="flex flex-col space-y-4 items-center">
+            <input type="hidden" value="<%= Phoenix.Controller.get_csrf_token() %>" name="_csrf_token"/>
+            <input type="text" name="secret" class="input" placeholder="<%= if @type == :token, do: "Token", else: "Password" %>"" />
             <button type="submit" class="button button-blue">
```

Comment:
I think it would be best to have two templates, but extract the common outer elements into ErrorLayout. This can definitely be refactored in a separate PR though!

Reply:
The conditional is fine by me. What I would try to do instead is to have a single conditional at the top what wraps everything instead trying to build everything dynamically. :)